// evergreen-ui/commonjs/colors/src/colors.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = {
  neutral: {
    '3A': 'rgba(67,90,111,0.025)',
    '5A': 'rgba(67,90,111,0.041)',
    '7A': 'rgba(67,90,111,0.057)',
    '10A': 'rgba(67,90,111,0.079)',
    '15A': 'rgba(67,90,111,0.114)',
    '20A': 'rgba(67,90,111,0.146)',
    '30A': 'rgba(67,90,111,0.204)',
    '40A': 'rgba(67,90,111,0.255)',
    '50A': 'rgba(67,90,111,0.301)',
    '60A': 'rgba(67,90,111,0.342)',
    '70A': 'rgba(67,90,111,0.38)',
    '80A': 'rgba(67,90,111,0.415)',
    '90A': 'rgba(67,90,111,0.447)',
    '100A': 'rgba(67,90,111,0.477)',
    '125A': 'rgba(67,90,111,0.544)',
    '150A': 'rgba(67,90,111,0.602)',
    '175A': 'rgba(67,90,111,0.653)',
    '200A': 'rgba(67,90,111,0.699)',
    '300A': 'rgba(67,90,111,0.845)',
    '400A': 'rgba(67,90,111,0.954)',
    '3': '#fafbfc',
    '5': '#f7f8fa',
    '7': '#f3f6f8',
    '10': '#eff2f5',
    '15': '#e7ecf1',
    '20': '#e1e7ed',
    '30': '#d5dee6',
    '40': '#cad5df',
    '50': '#c1ced9',
    '60': '#b8c7d4',
    '70': '#b0c1d0',
    '80': '#a9bbcb',
    '90': '#a2b6c7',
    '100': '#9cb1c3',
    '125': '#8ea6bb',
    '150': '#829cb4',
    '175': '#7894ad',
    '200': '#6e8ca8',
    '300': '#56748f',
    '400': '#496279',
    '500': '#435a6f',
    '600': '#37536c',
    '700': '#2c4b68',
    '800': '#1f4160',
    '900': '#153656',
    '1000': '#092744'
  },
  red: {
    '3A': 'rgba(243,99,49,0.025)',
    '5A': 'rgba(243,99,49,0.041)',
    '7A': 'rgba(243,99,49,0.057)',
    '10A': 'rgba(243,99,49,0.079)',
    '15A': 'rgba(243,99,49,0.114)',
    '20A': 'rgba(243,99,49,0.146)',
    '30A': 'rgba(243,99,49,0.204)',
    '40A': 'rgba(243,99,49,0.255)',
    '50A': 'rgba(243,99,49,0.301)',
    '60A': 'rgba(243,99,49,0.342)',
    '70A': 'rgba(243,99,49,0.38)',
    '80A': 'rgba(243,99,49,0.415)',
    '90A': 'rgba(243,99,49,0.447)',
    '100A': 'rgba(243,99,49,0.477)',
    '125A': 'rgba(243,99,49,0.544)',
    '150A': 'rgba(243,99,49,0.602)',
    '175A': 'rgba(243,99,49,0.653)',
    '200A': 'rgba(243,99,49,0.699)',
    '300A': 'rgba(243,99,49,0.845)',
    '400A': 'rgba(243,99,49,0.954)',
    '3': '#fffbfa',
    '5': '#fff9f7',
    '7': '#fef6f3',
    '10': '#fef3ef',
    '15': '#feede8',
    '20': '#fde8e1',
    '30': '#fddfd5',
    '40': '#fcd7ca',
    '50': '#fbd0c1',
    '60': '#fbcab9',
    '70': '#fac4b1',
    '80': '#fabeaa',
    '90': '#fab9a3',
    '100': '#f9b59d',
    '125': '#f8aa8f',
    '150': '#f8a183',
    '175': '#f79978',
    '200': '#f7926f',
    '300': '#f57b51',
    '400': '#f46a3a',
    '500': '#f36331',
    '600': '#f4541d',
    '700': '#f4450a',
    '800': '#dd3c06',
    '900': '#c63403',
    '1000': '#a32800'
  },
  green: {
    '3A': 'rgba(71,184,129,0.025)',
    '5A': 'rgba(71,184,129,0.041)',
    '7A': 'rgba(71,184,129,0.057)',
    '10A': 'rgba(71,184,129,0.079)',
    '15A': 'rgba(71,184,129,0.114)',
    '20A': 'rgba(71,184,129,0.146)',
    '30A': 'rgba(71,184,129,0.204)',
    '40A': 'rgba(71,184,129,0.255)',
    '50A': 'rgba(71,184,129,0.301)',
    '60A': 'rgba(71,184,129,0.342)',
    '70A': 'rgba(71,184,129,0.38)',
    '80A': 'rgba(71,184,129,0.415)',
    '90A': 'rgba(71,184,129,0.447)',
    '100A': 'rgba(71,184,129,0.477)',
    '125A': 'rgba(71,184,129,0.544)',
    '150A': 'rgba(71,184,129,0.602)',
    '175A': 'rgba(71,184,129,0.653)',
    '200A': 'rgba(71,184,129,0.699)',
    '300A': 'rgba(71,184,129,0.845)',
    '400A': 'rgba(71,184,129,0.954)',
    '3': '#fafdfc',
    '5': '#f7fcfa',
    '7': '#f5fbf8',
    '10': '#f0f9f5',
    '15': '#eaf7f1',
    '20': '#e4f5ed',
    '30': '#d9f1e5',
    '40': '#d0eddf',
    '50': '#c8ead9',
    '60': '#c0e7d4',
    '70': '#b9e4cf',
    '80': '#b3e2cb',
    '90': '#addfc7',
    '100': '#a7ddc3',
    '125': '#9bd8ba',
    '150': '#90d4b3',
    '175': '#87d1ad',
    '200': '#7ecda7',
    '300': '#64c395',
    '400': '#4fbb87',
    '500': '#47b881',
    '600': '#3faf77',
    '700': '#37a56d',
    '800': '#2d9760',
    '900': '#248953',
    '1000': '#197140'
  },
  blue: {
    '3A': 'rgba(1,108,209,0.025)',
    '5A': 'rgba(1,108,209,0.041)',
    '7A': 'rgba(1,108,209,0.057)',
    '10A': 'rgba(1,108,209,0.079)',
    '15A': 'rgba(1,108,209,0.114)',
    '20A': 'rgba(1,108,209,0.146)',
    '30A': 'rgba(1,108,209,0.204)',
    '40A': 'rgba(1,108,209,0.255)',
    '50A': 'rgba(1,108,209,0.301)',
    '60A': 'rgba(1,108,209,0.342)',
    '70A': 'rgba(1,108,209,0.38)',
    '80A': 'rgba(1,108,209,0.415)',
    '90A': 'rgba(1,108,209,0.447)',
    '100A': 'rgba(1,108,209,0.477)',
    '125A': 'rgba(1,108,209,0.544)',
    '150A': 'rgba(1,108,209,0.602)',
    '175A': 'rgba(1,108,209,0.653)',
    '200A': 'rgba(1,108,209,0.699)',
    '300A': 'rgba(1,108,209,0.845)',
    '400A': 'rgba(1,108,209,0.954)',
    '3': '#f8fbff',
    '5': '#f3f9ff',
    '7': '#eef7ff',
    '10': '#e7f3ff',
    '15': '#ddeeff',
    '20': '#d3eaff',
    '30': '#c2e1ff',
    '40': '#b3daff',
    '50': '#a5d3ff',
    '60': '#99cdff',
    '70': '#8ec8fe',
    '80': '#83c3fe',
    '90': '#7abefe',
    '100': '#71b9fe',
    '125': '#5db0fe',
    '150': '#4ba7fe',
    '175': '#3ca0fe',
    '200': '#2e99fe',
    '300': '#0384fe',
    '400': '#0173df',
    '500': '#016cd1',
    '600': '#0165c7',
    '700': '#015ebd',
    '800': '#0055b0',
    '900': '#004ca3',
    '1000': '#00408f'
  },
  purple: {
    '3A': 'rgba(104,84,192,0.025)',
    '5A': 'rgba(104,84,192,0.041)',
    '7A': 'rgba(104,84,192,0.057)',
    '10A': 'rgba(104,84,192,0.079)',
    '15A': 'rgba(104,84,192,0.114)',
    '20A': 'rgba(104,84,192,0.146)',
    '30A': 'rgba(104,84,192,0.204)',
    '40A': 'rgba(104,84,192,0.255)',
    '50A': 'rgba(104,84,192,0.301)',
    '60A': 'rgba(104,84,192,0.342)',
    '70A': 'rgba(104,84,192,0.38)',
    '80A': 'rgba(104,84,192,0.415)',
    '90A': 'rgba(104,84,192,0.447)',
    '100A': 'rgba(104,84,192,0.477)',
    '125A': 'rgba(104,84,192,0.544)',
    '150A': 'rgba(104,84,192,0.602)',
    '175A': 'rgba(104,84,192,0.653)',
    '200A': 'rgba(104,84,192,0.699)',
    '300A': 'rgba(104,84,192,0.845)',
    '400A': 'rgba(104,84,192,0.954)',
    '3': '#fbfbfd',
    '5': '#f9f8fc',
    '7': '#f6f5fb',
    '10': '#f3f1fa',
    '15': '#eeecf8',
    '20': '#e9e6f6',
    '30': '#e0dcf2',
    '40': '#d8d3ef',
    '50': '#d2ccec',
    '60': '#cbc5e9',
    '70': '#c6bee7',
    '80': '#c0b8e5',
    '90': '#bcb3e3',
    '100': '#b7ade1',
    '125': '#ada2dd',
    '150': '#a498d9',
    '175': '#9c8fd6',
    '200': '#9587d3',
    '300': '#7f6fca',
    '400': '#6f5cc3',
    '500': '#6854c0',
    '600': '#5d47bd',
    '700': '#553eb5',
    '800': '#4c37a8',
    '900': '#443099',
    '1000': '#382683'
  },
  pink: {
    '3A': 'rgba(230,56,113,0.025)',
    '5A': 'rgba(230,56,113,0.041)',
    '7A': 'rgba(230,56,113,0.057)',
    '10A': 'rgba(230,56,113,0.079)',
    '15A': 'rgba(230,56,113,0.114)',
    '20A': 'rgba(230,56,113,0.146)',
    '30A': 'rgba(230,56,113,0.204)',
    '40A': 'rgba(230,56,113,0.255)',
    '50A': 'rgba(230,56,113,0.301)',
    '60A': 'rgba(230,56,113,0.342)',
    '70A': 'rgba(230,56,113,0.38)',
    '80A': 'rgba(230,56,113,0.415)',
    '90A': 'rgba(230,56,113,0.447)',
    '100A': 'rgba(230,56,113,0.477)',
    '125A': 'rgba(230,56,113,0.544)',
    '150A': 'rgba(230,56,113,0.602)',
    '175A': 'rgba(230,56,113,0.653)',
    '200A': 'rgba(230,56,113,0.699)',
    '300A': 'rgba(230,56,113,0.845)',
    '400A': 'rgba(230,56,113,0.954)',
    '3': '#fefafb',
    '5': '#fef7f9',
    '7': '#fef4f7',
    '10': '#fdeff4',
    '15': '#fce8ef',
    '20': '#fbe2ea',
    '30': '#fad6e2',
    '40': '#f9ccdb',
    '50': '#f7c3d4',
    '60': '#f6bbce',
    '70': '#f5b3c9',
    '80': '#f5acc4',
    '90': '#f4a6c0',
    '100': '#f3a0bb',
    '125': '#f193b2',
    '150': '#f087aa',
    '175': '#ef7da2',
    '200': '#ee749c',
    '300': '#ea5787',
    '400': '#e74178',
    '500': '#e63871',
    '600': '#e42b68',
    '700': '#e31e5e',
    '800': '#d31a57',
    '900': '#c11850',
    '1000': '#a71545'
  },
  yellow: {
    '3A': 'rgba(247,198,83,0.025)',
    '5A': 'rgba(247,198,83,0.041)',
    '7A': 'rgba(247,198,83,0.057)',
    '10A': 'rgba(247,198,83,0.079)',
    '15A': 'rgba(247,198,83,0.114)',
    '20A': 'rgba(247,198,83,0.146)',
    '30A': 'rgba(247,198,83,0.204)',
    '40A': 'rgba(247,198,83,0.255)',
    '50A': 'rgba(247,198,83,0.301)',
    '60A': 'rgba(247,198,83,0.342)',
    '70A': 'rgba(247,198,83,0.38)',
    '80A': 'rgba(247,198,83,0.415)',
    '90A': 'rgba(247,198,83,0.447)',
    '100A': 'rgba(247,198,83,0.477)',
    '125A': 'rgba(247,198,83,0.544)',
    '150A': 'rgba(247,198,83,0.602)',
    '175A': 'rgba(247,198,83,0.653)',
    '200A': 'rgba(247,198,83,0.699)',
    '300A': 'rgba(247,198,83,0.845)',
    '400A': 'rgba(247,198,83,0.954)',
    '3': '#fffefb',
    '5': '#fffdf8',
    '7': '#fffcf5',
    '10': '#fefaf1',
    '15': '#fef9eb',
    '20': '#fef7e6',
    '30': '#fdf3dc',
    '40': '#fdf0d3',
    '50': '#fdeecb',
    '60': '#fcecc4',
    '70': '#fce9be',
    '80': '#fce7b8',
    '90': '#fbe6b2',
    '100': '#fbe4ad',
    '125': '#fbe0a1',
    '150': '#fadd97',
    '175': '#fada8f',
    '200': '#f9d787',
    '300': '#f8cf6e',
    '400': '#f7c95b',
    '500': '#f7c653',
    '600': '#f6b94b',
    '700': '#f4ab43',
    '800': '#f29738',
    '900': '#f0822d',
    '1000': '#ec601e'
  },
  turquoise: {
    '3A': 'rgba(20,181,208,0.025)',
    '5A': 'rgba(20,181,208,0.041)',
    '7A': 'rgba(20,181,208,0.057)',
    '10A': 'rgba(20,181,208,0.079)',
    '15A': 'rgba(20,181,208,0.114)',
    '20A': 'rgba(20,181,208,0.146)',
    '30A': 'rgba(20,181,208,0.204)',
    '40A': 'rgba(20,181,208,0.255)',
    '50A': 'rgba(20,181,208,0.301)',
    '60A': 'rgba(20,181,208,0.342)',
    '70A': 'rgba(20,181,208,0.38)',
    '80A': 'rgba(20,181,208,0.415)',
    '90A': 'rgba(20,181,208,0.447)',
    '100A': 'rgba(20,181,208,0.477)',
    '125A': 'rgba(20,181,208,0.544)',
    '150A': 'rgba(20,181,208,0.602)',
    '175A': 'rgba(20,181,208,0.653)',
    '200A': 'rgba(20,181,208,0.699)',
    '300A': 'rgba(20,181,208,0.845)',
    '400A': 'rgba(20,181,208,0.954)',
    '3': '#f9fefe',
    '5': '#f4fdfe',
    '7': '#f0fcfe',
    '10': '#ebfafd',
    '15': '#e2f8fc',
    '20': '#d9f7fb',
    '30': '#cbf3fa',
    '40': '#bdf0f9',
    '50': '#b2eef8',
    '60': '#a7ebf7',
    '70': '#9de9f6',
    '80': '#94e7f5',
    '90': '#8ce5f4',
    '100': '#84e3f3',
    '125': '#73dff2',
    '150': '#64dcf0',
    '175': '#57d9ef',
    '200': '#4bd6ee',
    '300': '#26ceea',
    '400': '#15bfdc',
    '500': '#14b5d0',
    '600': '#11aac6',
    '700': '#0fa0bc',
    '800': '#0b92ae',
    '900': '#0984a0',
    '1000': '#056f8a'
  },
  white: {
    '3A': 'rgba(255,255,255,0.025)',
    '5A': 'rgba(255,255,255,0.041)',
    '7A': 'rgba(255,255,255,0.057)',
    '10A': 'rgba(255,255,255,0.079)',
    '15A': 'rgba(255,255,255,0.114)',
    '20A': 'rgba(255,255,255,0.146)',
    '30A': 'rgba(255,255,255,0.204)',
    '40A': 'rgba(255,255,255,0.255)',
    '50A': 'rgba(255,255,255,0.301)',
    '60A': 'rgba(255,255,255,0.342)',
    '70A': 'rgba(255,255,255,0.38)',
    '80A': 'rgba(255,255,255,0.415)',
    '90A': 'rgba(255,255,255,0.447)',
    '100A': 'rgba(255,255,255,0.477)',
    '125A': 'rgba(255,255,255,0.544)',
    '150A': 'rgba(255,255,255,0.602)',
    '175A': 'rgba(255,255,255,0.653)',
    '200A': 'rgba(255,255,255,0.699)',
    '300A': 'rgba(255,255,255,0.845)',
    '400A': 'rgba(255,255,255,0.954)',
    '500': '#ffffff'
  }
};
