// evergreen-ui/esm/select-menu/src/Option.js
var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Pane } from '../../layers';
import { TableRow, TextTableCell } from '../../table';

var CheckIcon = function CheckIcon(_ref) {
  var _ref$fill = _ref.fill,
      fill = _ref$fill === undefined ? 'currentColor' : _ref$fill,
      props = _objectWithoutProperties(_ref, ['fill']);

  return React.createElement(
    'svg',
    _extends({ width: 10, height: 7, viewBox: '0 0 10 7' }, props),
    React.createElement('path', {
      fill: fill,
      fillRule: 'evenodd',
      d: 'M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z'
    })
  );
};

CheckIcon.propTypes = {
  fill: PropTypes.string
};

var Option = function (_PureComponent) {
  _inherits(Option, _PureComponent);

  function Option() {
    _classCallCheck(this, Option);

    return _possibleConstructorReturn(this, (Option.__proto__ || Object.getPrototypeOf(Option)).apply(this, arguments));
  }

  _createClass(Option, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          label = _props.label,
          onSelect = _props.onSelect,
          onDeselect = _props.onDeselect,
          isHighlighted = _props.isHighlighted,
          isSelected = _props.isSelected,
          style = _props.style,
          height = _props.height,
          props = _objectWithoutProperties(_props, ['label', 'onSelect', 'onDeselect', 'isHighlighted', 'isSelected', 'style', 'height']);

      return React.createElement(
        TableRow,
        _extends({
          isSelectable: true,
          isHighlighted: isHighlighted,
          onSelect: onSelect,
          onDeselect: onDeselect,
          isSelected: isSelected,
          style: style,
          display: 'flex',
          alignItems: 'center'
        }, props),
        React.createElement(
          Pane,
          {
            paddingLeft: 11,
            paddingRight: 8,
            marginTop: -5,
            opacity: isSelected ? 1 : 0,
            flexGrow: 0
          },
          React.createElement(CheckIcon, null)
        ),
        React.createElement(
          TextTableCell,
          {
            height: height,
            textProps: isSelected ? {
              color: 'inherit'
            } : {},
            paddingLeft: 0,
            borderRight: null,
            flex: 1,
            alignSelf: 'stretch'
          },
          label
        )
      );
    }
  }]);

  return Option;
}(PureComponent);

Option.propTypes = {
  label: PropTypes.string,
  style: PropTypes.any,
  height: PropTypes.number,
  onSelect: PropTypes.func,
  onDeselect: PropTypes.func,
  isHighlighted: PropTypes.bool,
  isSelected: PropTypes.bool
};
export default Option;
