// evergreen-ui/commonjs/table/src/TextTableCell.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _typography = require('../../typography');

var _TableCell = require('./TableCell');

var _TableCell2 = _interopRequireDefault(_TableCell);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ellipsis = {
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
};

var TextTableCell = function (_PureComponent) {
  _inherits(TextTableCell, _PureComponent);

  function TextTableCell() {
    _classCallCheck(this, TextTableCell);

    return _possibleConstructorReturn(this, (TextTableCell.__proto__ || Object.getPrototypeOf(TextTableCell)).apply(this, arguments));
  }

  _createClass(TextTableCell, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          textProps = _props.textProps,
          isNumber = _props.isNumber,
          props = _objectWithoutProperties(_props, ['children', 'textProps', 'isNumber']);

      return _react2.default.createElement(
        _TableCell2.default,
        props,
        _react2.default.createElement(
          _typography.Text,
          _extends({
            size: 300,
            flex: '1'
          }, ellipsis, isNumber ? {
            textAlign: 'right',
            fontFamily: 'mono'
          } : {}, textProps),
          children
        )
      );
    }
  }]);

  return TextTableCell;
}(_react.PureComponent);

TextTableCell.propTypes = _extends({}, _TableCell2.default.propTypes, {

  /**
   * Adds textAlign: right and fontFamily: mono.
   */
  isNumber: _propTypes2.default.bool.isRequired,

  /**
   * Pass additional props to the Text component.
   */
  textProps: _propTypes2.default.shape(_typography.Text.propTypes)
});
TextTableCell.defaultProps = {
  isNumber: false
};
exports.default = TextTableCell;
