// evergreen-ui/esm/textarea/src/Textarea.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import { TextStyles } from '../../typography';
import { InputAppearances, getBorderRadiusForTextSize } from '../../shared-styles';

var Textarea = function (_PureComponent) {
  _inherits(Textarea, _PureComponent);

  function Textarea() {
    _classCallCheck(this, Textarea);

    return _possibleConstructorReturn(this, (Textarea.__proto__ || Object.getPrototypeOf(Textarea)).apply(this, arguments));
  }

  _createClass(Textarea, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          css = _props.css,
          disabled = _props.disabled,
          textSize = _props.textSize,
          isInvalid = _props.isInvalid,
          spellCheck = _props.spellCheck,
          props = _objectWithoutProperties(_props, ['css', 'disabled', 'textSize', 'isInvalid', 'spellCheck']);

      var textStyle = TextStyles[textSize];
      var appearanceStyle = InputAppearances.default;
      var borderRadius = getBorderRadiusForTextSize({ textSize: textSize });

      return React.createElement(Box, _extends({
        is: 'textarea',
        disabled: disabled,
        borderRadius: borderRadius,
        spellCheck: spellCheck
      }, isInvalid ? { 'aria-invalid': true } : {}, disabled ? { color: 'extraMuted' } : {}, textStyle, {
        css: _extends({}, css, appearanceStyle)
      }, Textarea.styles, props));
    }
  }]);

  return Textarea;
}(PureComponent);

Textarea.propTypes = _extends({}, Box.propTypes, {
  disabled: PropTypes.bool.isRequired,
  isInvalid: PropTypes.bool.isRequired,
  spellCheck: PropTypes.bool.isRequired,
  textSize: PropTypes.oneOf(Object.keys(TextStyles))
});
Textarea.defaultProps = {
  disabled: false,
  isInvalid: false,
  spellCheck: true,
  textSize: 300
};
Textarea.styles = {
  boxSizing: 'border-box',
  minHeight: 80,
  minWidth: 280,
  paddingX: 10,
  paddingY: 8
};
export default Textarea;
