// evergreen-ui/esm/select/src/Select.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import { Text } from '../../typography';
import { TriangleIcon } from '../../icons';
import { getIconSizeForControlHeight, getBorderRadiusForControlHeight, getTextSizeForControlHeight } from '../../shared-styles';
import SelectAppearances from './styles/SelectAppearances';

var Select = function (_PureComponent) {
  _inherits(Select, _PureComponent);

  function Select() {
    _classCallCheck(this, Select);

    return _possibleConstructorReturn(this, (Select.__proto__ || Object.getPrototypeOf(Select)).apply(this, arguments));
  }

  _createClass(Select, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          name = _props.name,
          height = _props.height,
          children = _props.children,
          disabled = _props.disabled,
          onChange = _props.onChange,
          value = _props.value,
          required = _props.required,
          autofocus = _props.autofocus,
          isInvalid = _props.isInvalid,
          appearance = _props.appearance,
          props = _objectWithoutProperties(_props, ['id', 'name', 'height', 'children', 'disabled', 'onChange', 'value', 'required', 'autofocus', 'isInvalid', 'appearance']);

      var appearanceStyle = SelectAppearances[appearance];
      var textSize = getTextSizeForControlHeight({ height: height });
      var borderRadius = getBorderRadiusForControlHeight({ height: height });
      var iconSize = getIconSizeForControlHeight({ height: height });

      return React.createElement(
        Box,
        _extends({
          display: 'inline-flex',
          flex: 1,
          position: 'relative',
          width: 200,
          height: height
        }, props),
        React.createElement(
          Text,
          _extends({
            is: 'select',
            id: id,
            name: name,
            onChange: onChange,
            value: value
          }, required ? { required: true } : {}, autofocus ? { autofocus: true } : {}, disabled ? { disabled: true } : {}, isInvalid ? { 'aria-invalid': true } : {}, {
            css: appearanceStyle,
            size: textSize,
            borderRadius: borderRadius,
            textTransform: 'default',
            paddingLeft: Math.round(height / 3.2)
          }),
          children
        ),
        React.createElement(TriangleIcon, {
          height: height,
          aim: 'down',
          padding: 10,
          boxSizing: 'border-box',
          position: 'absolute',
          right: height >= 36 ? 4 : 0,
          color: disabled ? 'disabled' : 'default',
          iconSize: iconSize,
          css: {
            pointerEvents: 'none'
          }
        })
      );
    }
  }]);

  return Select;
}(PureComponent);

Select.propTypes = _extends({}, Box.propTypes, {
  id: PropTypes.string,
  name: PropTypes.string,
  children: PropTypes.node,
  onChange: PropTypes.func,
  value: PropTypes.any,
  appearance: PropTypes.oneOf(Object.keys(SelectAppearances)),
  required: PropTypes.bool,
  autofocus: PropTypes.bool,
  isInvalid: PropTypes.bool
});
Select.defaultProps = {
  appearance: 'default',
  height: 32
};
export default Select;
