// evergreen-ui/commonjs/select/src/Select.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uiBox = require('ui-box');

var _uiBox2 = _interopRequireDefault(_uiBox);

var _typography = require('../../typography');

var _icons = require('../../icons');

var _sharedStyles = require('../../shared-styles');

var _SelectAppearances = require('./styles/SelectAppearances');

var _SelectAppearances2 = _interopRequireDefault(_SelectAppearances);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Select = function (_PureComponent) {
  _inherits(Select, _PureComponent);

  function Select() {
    _classCallCheck(this, Select);

    return _possibleConstructorReturn(this, (Select.__proto__ || Object.getPrototypeOf(Select)).apply(this, arguments));
  }

  _createClass(Select, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          name = _props.name,
          height = _props.height,
          children = _props.children,
          disabled = _props.disabled,
          onChange = _props.onChange,
          value = _props.value,
          required = _props.required,
          autofocus = _props.autofocus,
          isInvalid = _props.isInvalid,
          appearance = _props.appearance,
          props = _objectWithoutProperties(_props, ['id', 'name', 'height', 'children', 'disabled', 'onChange', 'value', 'required', 'autofocus', 'isInvalid', 'appearance']);

      var appearanceStyle = _SelectAppearances2.default[appearance];
      var textSize = (0, _sharedStyles.getTextSizeForControlHeight)({ height: height });
      var borderRadius = (0, _sharedStyles.getBorderRadiusForControlHeight)({ height: height });
      var iconSize = (0, _sharedStyles.getIconSizeForControlHeight)({ height: height });

      return _react2.default.createElement(
        _uiBox2.default,
        _extends({
          display: 'inline-flex',
          flex: 1,
          position: 'relative',
          width: 200,
          height: height
        }, props),
        _react2.default.createElement(
          _typography.Text,
          _extends({
            is: 'select',
            id: id,
            name: name,
            onChange: onChange,
            value: value
          }, required ? { required: true } : {}, autofocus ? { autofocus: true } : {}, disabled ? { disabled: true } : {}, isInvalid ? { 'aria-invalid': true } : {}, {
            css: appearanceStyle,
            size: textSize,
            borderRadius: borderRadius,
            textTransform: 'default',
            paddingLeft: Math.round(height / 3.2)
          }),
          children
        ),
        _react2.default.createElement(_icons.TriangleIcon, {
          height: height,
          aim: 'down',
          padding: 10,
          boxSizing: 'border-box',
          position: 'absolute',
          right: height >= 36 ? 4 : 0,
          color: disabled ? 'disabled' : 'default',
          iconSize: iconSize,
          css: {
            pointerEvents: 'none'
          }
        })
      );
    }
  }]);

  return Select;
}(_react.PureComponent);

Select.propTypes = _extends({}, _uiBox2.default.propTypes, {
  id: _propTypes2.default.string,
  name: _propTypes2.default.string,
  children: _propTypes2.default.node,
  onChange: _propTypes2.default.func,
  value: _propTypes2.default.any,
  appearance: _propTypes2.default.oneOf(Object.keys(_SelectAppearances2.default)),
  required: _propTypes2.default.bool,
  autofocus: _propTypes2.default.bool,
  isInvalid: _propTypes2.default.bool
});
Select.defaultProps = {
  appearance: 'default',
  height: 32
};
exports.default = Select;
