// evergreen-ui/commonjs/typography/src/styles/LinkAppearances.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _colors = require('../../../colors');

var LinkAppearances = {
  blue: {
    color: _colors.colors.blue['500'],
    textDecoration: 'none',
    '&:hover': {
      color: _colors.colors.blue['300'],
      textDecoration: 'underline'
    },
    '&:active': {
      color: _colors.colors.blue['900'],
      textDecoration: 'none'
    },
    '&:focus': {
      boxShadow: '0 0 0 2px ' + _colors.colors.blue['100A']
    }
  },
  green: {
    color: _colors.colors.green['500'],
    textDecoration: 'none',
    '&:hover': {
      color: _colors.colors.green['300'],
      textDecoration: 'underline'
    },
    '&:active': {
      color: _colors.colors.green['900'],
      textDecoration: 'none'
    },
    '&:focus': {
      boxShadow: '0 0 0 2px ' + _colors.colors.green['100A']
    }
  },
  neutral: {
    color: _colors.colors.neutral['500'],
    textDecoration: 'none',
    '&:hover': {
      color: _colors.colors.neutral['300'],
      textDecoration: 'underline'
    },
    '&:active': {
      color: _colors.colors.neutral['900'],
      textDecoration: 'none'
    },
    '&:focus': {
      boxShadow: '0 0 0 2px ' + _colors.colors.neutral['100A']
    }
  }
};

exports.default = LinkAppearances;
