// evergreen-ui/commonjs/typography/src/SubHeading.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _lodash = require('lodash.mapvalues');

var _lodash2 = _interopRequireDefault(_lodash);

var _TextStyles = require('./styles/TextStyles');

var _TextStyles2 = _interopRequireDefault(_TextStyles);

var _Text = require('./Text');

var _Text2 = _interopRequireDefault(_Text);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var textStyleTransformation = function textStyleTransformation(_ref) {
  var fontWeight = _ref.fontWeight,
      textStyle = _objectWithoutProperties(_ref, ['fontWeight']);

  return _extends({}, textStyle, {
    fontWeight: fontWeight - 100
  });
};

var textStyles = (0, _lodash2.default)(_TextStyles2.default, textStyleTransformation);

var SubHeading = function (_PureComponent) {
  _inherits(SubHeading, _PureComponent);

  function SubHeading() {
    _classCallCheck(this, SubHeading);

    return _possibleConstructorReturn(this, (SubHeading.__proto__ || Object.getPrototypeOf(SubHeading)).apply(this, arguments));
  }

  _createClass(SubHeading, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(_Text2.default, _extends({
        is: 'h3',
        color: 'dark',
        fontFamily: 'display',
        marginTop: 0,
        marginBottom: 0,
        textStyles: textStyles
      }, this.props));
    }
  }]);

  return SubHeading;
}(_react.PureComponent);

SubHeading.propTypes = _extends({}, _Text2.default.propTypes);
exports.default = SubHeading;
