// evergreen-ui/commonjs/typography/src/Text.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uiBox = require('ui-box');

var _uiBox2 = _interopRequireDefault(_uiBox);

var _FontFamilies = require('./styles/FontFamilies');

var _FontFamilies2 = _interopRequireDefault(_FontFamilies);

var _TextStyles = require('./styles/TextStyles');

var _TextStyles2 = _interopRequireDefault(_TextStyles);

var _TextUppercaseStyles = require('./styles/TextUppercaseStyles');

var _TextUppercaseStyles2 = _interopRequireDefault(_TextUppercaseStyles);

var _TextColors = require('./styles/TextColors');

var _TextColors2 = _interopRequireDefault(_TextColors);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var isDev = process.env.NODE_ENV === 'development' || !process.env.NODE_ENV;

var Text = function (_PureComponent) {
  _inherits(Text, _PureComponent);

  function Text() {
    _classCallCheck(this, Text);

    return _possibleConstructorReturn(this, (Text.__proto__ || Object.getPrototypeOf(Text)).apply(this, arguments));
  }

  _createClass(Text, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          size = _props.size,
          color = _props.color,
          textStyles = _props.textStyles,
          textUppercaseStyles = _props.textUppercaseStyles,
          fontFamily = _props.fontFamily,
          isUppercase = _props.isUppercase,
          props = _objectWithoutProperties(_props, ['size', 'color', 'textStyles', 'textUppercaseStyles', 'fontFamily', 'isUppercase']);

      var textStyle = textStyles[size];

      if (isUppercase) {
        // Only 100 and 200 support uppercase styles atm
        // Fallback on non uppercase atm
        if (Object.prototype.hasOwnProperty.call(textUppercaseStyles, size)) {
          textStyle = textUppercaseStyles[size];
        } else if (isDev) {
          console.error('Uppercase ' + size + ' not supported. <Text isUppercase> only supports the following sizes: ' + JSON.stringify(Object.keys(textUppercaseStyles).map(Number)));
        }
      }

      return _react2.default.createElement(_uiBox2.default, _extends({
        is: 'span'
      }, color ? { color: _TextColors2.default[color] || color } : {}, {
        fontFamily: _FontFamilies2.default[fontFamily] || fontFamily
      }, textStyle, props));
    }
  }]);

  return Text;
}(_react.PureComponent);

Text.propTypes = _extends({}, _uiBox2.default.propTypes, {

  /**
   * Size of the text style.
   * Can be: 100, 200, 300, 400, 500, 600, 700, 800, 900.
   */
  size: _propTypes2.default.oneOf(Object.keys(_TextStyles2.default).map(Number)),

  /**
   * Font family.
   * Can be: ui, display or mono
   */
  fontFamily: _propTypes2.default.oneOf(Object.keys(_FontFamilies2.default)),

  /**
   * Sets the text to uppercase.
   * Only sizes 100 and 200 support uppercase styles at the moment.
   */
  isUppercase: _propTypes2.default.bool,

  /**
   * The text styles available.
   * This is overridden by other text components that implement Text as the base.
   * You should avoid setting this manually.
   */
  textStyles: _propTypes2.default.object,

  /**
   * The uppercase text styles.
   * You should avoid setting this manually.
   */
  textUppercaseStyles: _propTypes2.default.object
});
Text.defaultProps = {
  size: 500,
  color: 'default',
  fontFamily: 'ui',
  textStyles: _TextStyles2.default,
  textUppercaseStyles: _TextUppercaseStyles2.default
};
exports.default = Text;
