// evergreen-ui/commonjs/shared-styles/src/styles/CheckboxAppearances.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _colors = require('../../../colors');

var _controlBaseStyle = require('./controlBaseStyle');

var _controlBaseStyle2 = _interopRequireDefault(_controlBaseStyle);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var disabledState = '&[disabled] + div';
var hoverState = '&:not([disabled]):hover + div';
var focusState = '&:not([disabled]):focus + div';
var activeState = '&:not([disabled]):active + div';
var checkedState = '&:checked + div';
var checkedActiveState = '&:not([disabled]):checked:active + div';
var checkedDisabledState = '&[disabled]:checked + div';

var hiddenCheckboxStyle = {
  border: '0',
  clip: 'rect(1px, 1px, 1px, 1px)',
  height: '1',
  overflow: 'hidden',
  padding: '0',
  position: 'absolute',
  whiteSpace: 'nowrap',
  width: '1',
  opacity: '0'
};

var CheckboxAppearances = {
  default: _extends({}, hiddenCheckboxStyle, (_extends2 = {
    '& + div > svg': {
      display: 'none'
    },
    '& + div': _extends({}, _controlBaseStyle2.default, {
      color: 'white',
      backgroundColor: 'white',
      backgroundImage: 'linear-gradient(to top, ' + _colors.colors.neutral['5A'] + ', white)',
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['20A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['10A']
    })
  }, _defineProperty(_extends2, disabledState, {
    cursor: 'not-allowed',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['20A'],
    backgroundColor: _colors.colors.neutral['5A'],
    backgroundImage: 'none'
  }), _defineProperty(_extends2, hoverState, {
    backgroundImage: 'linear-gradient(to top, ' + _colors.colors.neutral['7A'] + ', ' + _colors.colors.neutral['3A'] + ' )',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['40A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['15A']
  }), _defineProperty(_extends2, focusState, {
    boxShadow: '0 0 0 2px ' + _colors.colors.blue['20A'] + ', inset 0 0 0 1px ' + _colors.colors.neutral['70A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['10A']
  }), _defineProperty(_extends2, activeState, {
    color: _colors.colors.blue['1000'],
    backgroundImage: 'none',
    backgroundColor: _colors.colors.blue['10A'],
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.blue['80A']
  }), _defineProperty(_extends2, checkedState, {
    backgroundColor: _colors.colors.blue['500'],
    color: 'white',
    backgroundImage: 'linear-gradient(to top, ' + _colors.colors.blue['600'] + ', ' + _colors.colors.blue['400'] + ')',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['30A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['30A'],
    '& > svg': {
      display: 'block'
    }
  }), _defineProperty(_extends2, checkedDisabledState, {
    color: _colors.colors.neutral['300'],
    backgroundColor: 'white',
    backgroundImage: 'linear-gradient(to top, ' + _colors.colors.neutral['5A'] + ', white)',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['20A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['10A'],
    '& > svg': {
      display: 'block'
    }
  }), _defineProperty(_extends2, checkedActiveState, {
    backgroundColor: _colors.colors.blue['500'],
    color: 'white',
    backgroundImage: 'linear-gradient(to top, ' + _colors.colors.blue['800'] + ', ' + _colors.colors.blue['900'] + ')',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['30A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['30A'],
    '& > svg': {
      display: 'block'
    }
  }), _extends2))
};

exports.default = CheckboxAppearances;
