// evergreen-ui/commonjs/shared-styles/src/styles/InputAppearances.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _colors = require('../../../colors');

var InputAppearances = {
  default: {
    WebkitAppearance: 'none',
    border: 'none',
    backgroundColor: 'white',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['30A'] + ', inset 0 1px 2px ' + _colors.colors.neutral['20A'],
    '&[aria-invalid="true"]': {
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.red['500'] + ', inset 0 1px 1px ' + _colors.colors.neutral['40A']
    },
    '&::placeholder': {
      color: _colors.colors.neutral['100A']
    },
    '&:focus': {
      zIndex: 2,
      outline: 'none',
      boxShadow: 'inset 0 0 2px ' + _colors.colors.neutral['40A'] + ', inset 0 0 0 1px ' + _colors.colors.blue['150A'] + ', 0 0 0 3px ' + _colors.colors.blue['15A']
    },
    '&:disabled': {
      cursor: 'not-allowed',
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['20A'],
      backgroundColor: _colors.colors.neutral['5A']
    }
  },
  neutral: {
    WebkitAppearance: 'none',
    border: 'none',
    backgroundColor: _colors.colors.neutral['10A'],
    '&[aria-invalid="true"]': {
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.red['500']
    },
    '&::placeholder': {
      color: _colors.colors.neutral['200A']
    },
    '&:focus': {
      outline: 'none',
      boxShadow: '0 0 0 2px ' + _colors.colors.blue['150A']
    },
    '&:disabled': {
      cursor: 'not-allowed',
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['20A'],
      backgroundColor: _colors.colors.neutral['5A']
    }
  }
};

exports.default = InputAppearances;
