// evergreen-ui/commonjs/segmented-control/src/SegmentedControl.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uiBox = require('ui-box');

var _uiBox2 = _interopRequireDefault(_uiBox);

var _SegmentedControlRadio = require('./SegmentedControlRadio');

var _SegmentedControlRadio2 = _interopRequireDefault(_SegmentedControlRadio);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var radioCount = 1; // Used for generating unique input names

var SegmentedControl = function (_PureComponent) {
  _inherits(SegmentedControl, _PureComponent);

  function SegmentedControl(props, context) {
    _classCallCheck(this, SegmentedControl);

    var _this = _possibleConstructorReturn(this, (SegmentedControl.__proto__ || Object.getPrototypeOf(SegmentedControl)).call(this, props, context));

    _this.handleChange = function (value) {
      // Save a render cycle when it's a controlled input
      if (!_this.props.value) {
        _this.setState({ value: value });
      }

      if (_this.props.onChange) {
        _this.props.onChange(value);
      }
    };

    _this.state = {
      value: props.defaultValue || props.options[0].value
    };

    _this.name = 'SegmentedControl-' + radioCount;
    radioCount += 1;
    return _this;
  }

  _createClass(SegmentedControl, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          name = _props.name,
          height = _props.height,
          options = _props.options,
          onChange = _props.onChange,
          defaultValue = _props.defaultValue,
          props = _objectWithoutProperties(_props, ['name', 'height', 'options', 'onChange', 'defaultValue']);

      // Allows it to behave like a controlled input


      var value = this.props.value || this.state.value;
      return _react2.default.createElement(
        _uiBox2.default,
        _extends({ display: 'flex', marginRight: -1, height: height }, props),
        options.map(function (option, index) {
          return _react2.default.createElement(_SegmentedControlRadio2.default, {
            key: option.value,
            name: name,
            label: option.label,
            value: option.value,
            height: height,
            checked: value === option.value,
            onChange: _this2.handleChange,
            appearance: 'default',
            isFirstItem: index === 0,
            isLastItem: index === options.length - 1
          });
        })
      );
    }
  }]);

  return SegmentedControl;
}(_react.PureComponent);

SegmentedControl.propTypes = _extends({}, _uiBox.spacing.propTypes, _uiBox.position.propTypes, _uiBox.layout.propTypes, _uiBox.dimensions.propTypes, {

  /**
   * The options for the radios of the Segmented Control.
   */
  options: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    label: _propTypes2.default.node.isRequired,
    value: _propTypes2.default.string.isRequired
  })).isRequired,

  /**
   * The current value of the Segmented Control when controlled.
   */
  value: _propTypes2.default.string,

  /**
   * The default value of the Segmented Control when uncontrolled.
   */
  defaultValue: _propTypes2.default.string,

  /**
   * Function called when the value changes.
   */
  onChange: _propTypes2.default.func,

  /**
   * The name of the radio group.
   */
  name: _propTypes2.default.string,

  /**
   * The height of the Segmented Control.
   */
  height: _propTypes2.default.number
});
SegmentedControl.defaultProps = {
  height: 32
};
exports.default = SegmentedControl;
