// evergreen-ui/commonjs/segmented-control/src/SegmentedControlRadio.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _css;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uiBox = require('ui-box');

var _uiBox2 = _interopRequireDefault(_uiBox);

var _colors = require('../../colors');

var _typography = require('../../typography');

var _sharedStyles = require('../../shared-styles');

var _SegmentedControlAppearances = require('./styles/SegmentedControlAppearances');

var _SegmentedControlAppearances2 = _interopRequireDefault(_SegmentedControlAppearances);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var keysSegmentedControlAppearances = Object.keys(_SegmentedControlAppearances2.default);

var labelClass = (0, _uiBox.css)({
  display: 'flex',
  flex: 1,
  alignItems: 'center',
  justifyContent: 'center',
  position: 'relative'
});

var wrapperClass = (0, _uiBox.css)((_css = {
  position: 'relative',
  display: 'flex',
  flex: 1,
  cursor: 'pointer',
  marginLeft: '-1px'
}, _defineProperty(_css, ':first-child .' + labelClass, {
  borderTopLeftRadius: 3,
  borderBottomLeftRadius: 3
}), _defineProperty(_css, ':last-child .' + labelClass, {
  borderTopRightRadius: 3,
  borderBottomRightRadius: 3
}), _css));

var offscreenCss = (0, _uiBox.css)({
  overflow: 'hidden',
  position: 'absolute',
  height: '1px',
  width: '1px',
  padding: 0,
  margin: '-1px',
  border: 0,
  clip: 'rect(0 0 0 0)'
});

var SegmentedControlRadio = function (_PureComponent) {
  _inherits(SegmentedControlRadio, _PureComponent);

  function SegmentedControlRadio() {
    _classCallCheck(this, SegmentedControlRadio);

    return _possibleConstructorReturn(this, (SegmentedControlRadio.__proto__ || Object.getPrototypeOf(SegmentedControlRadio)).apply(this, arguments));
  }

  _createClass(SegmentedControlRadio, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          name = _props.name,
          label = _props.label,
          value = _props.value,
          height = _props.height,
          checked = _props.checked,
          _onChange = _props.onChange,
          appearance = _props.appearance,
          isFirstItem = _props.isFirstItem,
          isLastItem = _props.isLastItem;


      var textSize = (0, _sharedStyles.getTextSizeForControlHeight)({ height: height });
      var borderRadius = (0, _sharedStyles.getBorderRadiusForControlHeight)({ height: height });
      var styles = _SegmentedControlAppearances2.default[appearance];

      return _react2.default.createElement(
        _uiBox2.default,
        _extends({
          is: 'label',
          className: '' + wrapperClass,
          css: styles
        }, checked ? {
          'data-active': true
        } : {}, isFirstItem ? {
          borderTopLeftRadius: borderRadius,
          borderBottomLeftRadius: borderRadius
        } : {}, isLastItem ? {
          borderTopRightRadius: borderRadius,
          borderBottomRightRadius: borderRadius
        } : {}),
        _react2.default.createElement('input', {
          type: 'radio',
          className: '' + offscreenCss,
          name: name,
          value: value,
          checked: checked,
          onChange: function onChange(e) {
            return _onChange(e.target.value);
          }
        }),
        _react2.default.createElement(
          _typography.Text,
          _extends({
            fontWeight: 500,
            size: textSize,
            className: '' + labelClass
          }, checked ? { color: _colors.colors.blue['500'] } : {}),
          label
        )
      );
    }
  }]);

  return SegmentedControlRadio;
}(_react.PureComponent);

SegmentedControlRadio.propTypes = {
  /**
   * The name attribute of the radio input.
   */
  name: _propTypes2.default.string.isRequired,

  /**
   * The label used for the radio.
   */
  label: _propTypes2.default.node.isRequired,

  /**
   * The value attribute of the radio input.
   */
  value: _propTypes2.default.string.isRequired,

  /**
   * The height of the control.
   */
  height: _propTypes2.default.number.isRequired,

  /**
   * When true, the radio input is checked.
   */
  checked: _propTypes2.default.bool.isRequired,

  /**
   * Function called when the state changes.
   */
  onChange: _propTypes2.default.func.isRequired,

  /**
   * The appearance of the control. Currently only `default` is possible.
   */
  appearance: _propTypes2.default.oneOf(keysSegmentedControlAppearances).isRequired,

  /**
   * When true, this item is the first item.
   */
  isFirstItem: _propTypes2.default.bool,

  /**
   * When true, this item is the last item.
   */
  isLastItem: _propTypes2.default.bool
};
exports.default = SegmentedControlRadio;
