// evergreen-ui/esm/avatar/src/Avatar.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import { Image } from '../../image';
import { Text } from '../../typography';
import { FillAppearances } from '../../shared-styles';
import { colors } from '../../colors';
import globalGetInitials from './utils/getInitials';
import globalHash from './utils/hash';

var keysFillAppearances = Object.keys(FillAppearances.default);

var initialsProps = {
  top: 0,
  position: 'absolute',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  lineHeight: 1
};

var getInitialsFontSize = function getInitialsFontSize(size, heightLimitOneCharacter) {
  if (size <= heightLimitOneCharacter) {
    return Math.ceil(size / 2.2);
  }
  return Math.ceil(size / 2.6);
};

var Avatar = function (_PureComponent) {
  _inherits(Avatar, _PureComponent);

  function Avatar(props, context) {
    _classCallCheck(this, Avatar);

    var _this = _possibleConstructorReturn(this, (Avatar.__proto__ || Object.getPrototypeOf(Avatar)).call(this, props, context));

    _this.handleError = function () {
      _this.setState({
        imageHasFailedLoading: true
      });
    };

    _this.state = {
      imageHasFailedLoading: false
    };
    return _this;
  }

  _createClass(Avatar, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          src = _props.src,
          size = _props.size,
          name = _props.name,
          hash = _props.hash,
          isSolid = _props.isSolid,
          propsHashValue = _props.hashValue,
          getInitials = _props.getInitials,
          useAutoColor = _props.useAutoColor,
          propsAppearance = _props.appearance,
          forceShowInitials = _props.forceShowInitials,
          heightLimitOneCharacter = _props.heightLimitOneCharacter,
          props = _objectWithoutProperties(_props, ['src', 'size', 'name', 'hash', 'isSolid', 'hashValue', 'getInitials', 'useAutoColor', 'appearance', 'forceShowInitials', 'heightLimitOneCharacter']);

      var imageHasFailedLoading = this.state.imageHasFailedLoading;

      var imageUnavailable = !src || imageHasFailedLoading;
      var initialsFontSize = getInitialsFontSize(size, heightLimitOneCharacter) + 'px';

      var initials = getInitials(name);
      if (size <= heightLimitOneCharacter) {
        initials = initials.substring(0, 1);
      }

      var hashValue = hash(propsHashValue || name);
      var appearanceKey = propsAppearance;
      if (appearanceKey === undefined) {
        appearanceKey = keysFillAppearances[hashValue % keysFillAppearances.length];
      }

      var appearance = FillAppearances[isSolid ? 'solid' : 'default'][appearanceKey];

      return React.createElement(
        Box,
        _extends({
          width: size,
          height: size,
          overflow: 'hidden',
          borderRadius: 9999,
          position: 'relative',
          display: 'inline-flex',
          justifyContent: 'center',
          backgroundColor: colors.neutral['300'],
          title: name
        }, appearance, props),
        (imageUnavailable || forceShowInitials) && React.createElement(
          Text,
          {
            css: initialsProps,
            fontSize: initialsFontSize,
            lineHeight: initialsFontSize,
            width: size,
            height: size,
            color: appearance.color
          },
          initials
        ),
        !imageUnavailable && React.createElement(Image, {
          width: 'auto',
          height: '100%',
          src: src,
          onError: this.handleError
        })
      );
    }
  }]);

  return Avatar;
}(PureComponent);

Avatar.propTypes = _extends({}, Image.propTypes, {
  size: PropTypes.number,
  name: PropTypes.string,
  // Hash value defaults to name
  hashValue: PropTypes.string,
  hash: PropTypes.func,
  isSolid: PropTypes.bool,
  appearance: PropTypes.oneOf(keysFillAppearances),
  getInitials: PropTypes.func,
  // In some cases Gravatar returns transparent pngs
  // we still want to see the initials
  forceShowInitials: PropTypes.bool,
  heightLimitOneCharacter: PropTypes.number
});
Avatar.defaultProps = {
  size: 24,
  hash: globalHash,
  isSolid: false,
  getInitials: globalGetInitials,
  forceShowInitials: false,
  heightLimitOneCharacter: 20
};
export default Avatar;
