// evergreen-ui/commonjs/text-input/src/TextInputField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uiBox = require('ui-box');

var _formField = require('../../form-field');

var _TextInput = require('./TextInput');

var _TextInput2 = _interopRequireDefault(_TextInput);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var idCounter = 0;

var TextInputField = function (_PureComponent) {
  _inherits(TextInputField, _PureComponent);

  function TextInputField(props) {
    _classCallCheck(this, TextInputField);

    var _this = _possibleConstructorReturn(this, (TextInputField.__proto__ || Object.getPrototypeOf(TextInputField)).call(this, props));

    _this.state = {
      id: (props.id || idCounter++).toString()
    };
    return _this;
  }

  _createClass(TextInputField, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          unusedId = _props.id,
          label = _props.label,
          description = _props.description,
          validationMessage = _props.validationMessage,
          inputHeight = _props.inputHeight,
          inputWidth = _props.inputWidth,
          disabled = _props.disabled,
          required = _props.required,
          isInvalid = _props.isInvalid,
          appearance = _props.appearance,
          placeholder = _props.placeholder,
          spellCheck = _props.spellCheck,
          props = _objectWithoutProperties(_props, ['id', 'label', 'description', 'validationMessage', 'inputHeight', 'inputWidth', 'disabled', 'required', 'isInvalid', 'appearance', 'placeholder', 'spellCheck']);

      var id = this.state.id;

      /**
       * Split the wrapper props from the input props.
       */

      var _splitBoxProps = (0, _uiBox.splitBoxProps)(props),
          matchedProps = _splitBoxProps.matchedProps,
          remainingProps = _splitBoxProps.remainingProps;

      return _react2.default.createElement(
        _formField.FormField,
        _extends({
          marginBottom: 24,
          label: label,
          isRequired: required,
          description: description,
          validationMessage: validationMessage,
          labelFor: id
        }, matchedProps),
        _react2.default.createElement(_TextInput2.default, _extends({
          id: id,
          width: inputWidth,
          height: inputHeight,
          disabled: disabled,
          required: required,
          isInvalid: isInvalid,
          appearance: appearance,
          placeholder: placeholder,
          spellCheck: spellCheck
        }, remainingProps))
      );
    }
  }]);

  return TextInputField;
}(_react.PureComponent);

TextInputField.propTypes = _extends({}, _TextInput2.default.propTypes, _formField.FormField.propTypes, {

  /**
   * The label used above the input element.
   */
  label: _propTypes2.default.node.isRequired,

  /**
   * Passed on the label as a htmlFor prop.
   */
  labelFor: _propTypes2.default.string,

  /**
   * Wether or not show a asterix after the label.
   */
  isRequired: _propTypes2.default.bool,

  /**
   * A optional description of the field under the input element.
   */
  description: _propTypes2.default.node,

  /**
   * If a validation message is passed it is shown under the input element
   * and above the description.
   */
  validationMessage: _propTypes2.default.node,

  /**
   * The height of the input element.
   */
  inputHeight: _propTypes2.default.number,

  /**
   * The width of the input width.
   */
  inputWidth: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string])
});
TextInputField.defaultProps = {
  /**
   * The input width should be as wide as the form field.
   */
  inputWidth: '100%',

  /**
   * Increase the height somewhat from 32 because we are dealing mainly
   * with traditional long style forms.
   */
  inputHeight: 36
};
exports.default = TextInputField;
