// evergreen-ui/esm/tabs/src/SidebarTab.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import Box from 'ui-box';
import { TriangleIcon } from '../../icons';
import Tab from './Tab';

var Tablist = function (_PureComponent) {
  _inherits(Tablist, _PureComponent);

  function Tablist() {
    _classCallCheck(this, Tablist);

    return _possibleConstructorReturn(this, (Tablist.__proto__ || Object.getPrototypeOf(Tablist)).apply(this, arguments));
  }

  _createClass(Tablist, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          height = _props.height,
          isSelected = _props.isSelected,
          props = _objectWithoutProperties(_props, ['children', 'height', 'isSelected']);

      return React.createElement(
        Tab,
        _extends({
          isSelected: isSelected,
          height: height
        }, Tablist.styles, props),
        React.createElement(
          Box,
          { is: 'span', flex: '1' },
          children
        ),
        isSelected && React.createElement(TriangleIcon, {
          height: height,
          width: height,
          iconSize: 14,
          aim: 'right',
          color: 'inherit'
        })
      );
    }
  }]);

  return Tablist;
}(PureComponent);

Tablist.propTypes = _extends({}, Tab.propTypes);
Tablist.defaultProps = {
  height: 32
};
Tablist.styles = {
  width: '100%',
  paddingX: 0,
  paddingLeft: 8,
  marginX: 0,
  marginBottom: 4,
  justifyContent: 'auto'
};
export default Tablist;
