// evergreen-ui/commonjs/form-field/src/FormFieldValidationMessage.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _typography = require('../../typography');

var _colors = require('../../colors');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var FormFieldValidationMessage = function (_PureComponent) {
  _inherits(FormFieldValidationMessage, _PureComponent);

  function FormFieldValidationMessage() {
    _classCallCheck(this, FormFieldValidationMessage);

    return _possibleConstructorReturn(this, (FormFieldValidationMessage.__proto__ || Object.getPrototypeOf(FormFieldValidationMessage)).apply(this, arguments));
  }

  _createClass(FormFieldValidationMessage, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(_typography.Paragraph, _extends({
        marginTop: 0,
        size: 300,
        color: _colors.colors.red['700']
      }, this.props, {
        role: 'alert'
      }));
    }
  }]);

  return FormFieldValidationMessage;
}(_react.PureComponent);

FormFieldValidationMessage.propTypes = _extends({}, _typography.Paragraph.propTypes);
exports.default = FormFieldValidationMessage;
