// evergreen-ui/commonjs/select/src/styles/SelectAppearances.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _colors = require('../../../colors');

var baseStyle = {
  MozAppearance: 'none',
  WebkitAppearance: 'none',
  border: 'none',
  flex: 1,
  background: 'none',
  width: '100%',
  WebkitFontSmoothing: 'antialiased',
  boxSizing: 'border-box',
  textDecoration: 'none',
  transition: 'box-shadow 80ms ease-in-out',
  outline: 'none',
  cursor: 'pointer',
  ':-moz-focusring': {
    color: 'transparent',
    textShadow: '0 0 0 #000'
  },
  '[disabled]': {
    cursor: 'not-allowed',
    opacity: 0.8,
    backgroundImage: 'none',
    backgroundColor: _colors.colors.neutral['10A'],
    boxShadow: 'none',
    color: _colors.colors.neutral['300A']
  },
  '&[aria-invalid]': {
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.red['500'] + ', inset 0 1px 2px ' + _colors.colors.neutral['40A']
  }
};

var SelectAppearances = {
  default: _extends({}, baseStyle, {
    backgroundColor: 'white',
    backgroundImage: 'linear-gradient(to top, ' + _colors.colors.neutral['5A'] + ', ' + _colors.colors.white['5A'] + ')',
    boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['20A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['10A'],
    '&:not([disabled]):hover': {
      backgroundImage: 'linear-gradient(to top, ' + _colors.colors.neutral['7A'] + ', ' + _colors.colors.neutral['3A'] + ' )',
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.neutral['40A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['15A']
    },
    '&:not([disabled]):focus': {
      boxShadow: '0 0 0 3px ' + _colors.colors.blue['20A'] + ', inset 0 0 0 1px ' + _colors.colors.neutral['70A'] + ', inset 0 -1px 1px 0 ' + _colors.colors.neutral['10A']
    },
    '&:not([disabled]):active': {
      color: _colors.colors.blue['1000'],
      backgroundImage: 'none',
      backgroundColor: _colors.colors.blue['10A'],
      boxShadow: 'inset 0 0 0 1px ' + _colors.colors.blue['80A']
    }
  })
};

exports.default = SelectAppearances;
