// evergreen-ui/esm/table/src/SearchTableHeaderCell.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text } from '../../typography';
import { SearchIcon } from '../../icons';
import { colors } from '../../colors';
import TableHeaderCell from './TableHeaderCell';

var invisibleInput = {
  border: 'none',
  backgroundColor: 'transparent',
  WebkitAppearance: 'none',
  WebkitFontSmoothing: 'antialiased',

  '&:focus': {
    outline: 'none'
  },

  '&::placeholder': {
    color: colors.neutral['100A']
  }
};

var SearchTableHeaderCell = function (_PureComponent) {
  _inherits(SearchTableHeaderCell, _PureComponent);

  function SearchTableHeaderCell() {
    _classCallCheck(this, SearchTableHeaderCell);

    return _possibleConstructorReturn(this, (SearchTableHeaderCell.__proto__ || Object.getPrototypeOf(SearchTableHeaderCell)).apply(this, arguments));
  }

  _createClass(SearchTableHeaderCell, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          value = _props.value,
          children = _props.children,
          _onChange = _props.onChange,
          autoFocus = _props.autoFocus,
          spellCheck = _props.spellCheck,
          placeholder = _props.placeholder,
          props = _objectWithoutProperties(_props, ['value', 'children', 'onChange', 'autoFocus', 'spellCheck', 'placeholder']);

      return React.createElement(
        TableHeaderCell,
        props,
        React.createElement(SearchIcon, { marginLeft: -8, marginTop: -0.5, iconSize: 12 }),
        React.createElement(Text, {
          is: 'input',
          size: 300,
          flex: '1',
          css: invisibleInput,
          value: value,
          onChange: function onChange(e) {
            return _onChange(e.target.value);
          },
          autoFocus: autoFocus,
          spellCheck: spellCheck,
          fontWeight: 500,
          marginLeft: -2,
          paddingLeft: 0,
          placeholder: placeholder
        })
      );
    }
  }]);

  return SearchTableHeaderCell;
}(PureComponent);

SearchTableHeaderCell.propTypes = _extends({}, TableHeaderCell.propTypes, {

  /**
   * The value of the input.
   */
  value: PropTypes.string,

  /**
   * Handler to be called when the input changes.
   */
  onChange: PropTypes.func,

  /**
   * Sets whether the component should be automatically focused on component render.
   */
  autoFocus: PropTypes.bool,

  /**
   * Sets whether to apply spell checking to the content.
   */
  spellCheck: PropTypes.bool,

  /**
   * Text to display in the input if the input is empty.
   */
  placeholder: PropTypes.string
});
SearchTableHeaderCell.defaultProps = {
  onChange: function onChange() {},
  spellCheck: true,
  placeholder: 'Filter...'
};
export default SearchTableHeaderCell;
