// evergreen-ui/esm/table/src/TableRow.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Pane } from '../../layers';
import { selectableRowStyle } from '../../shared-styles';

var TableRow = function (_PureComponent) {
  _inherits(TableRow, _PureComponent);

  function TableRow() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, TableRow);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = TableRow.__proto__ || Object.getPrototypeOf(TableRow)).call.apply(_ref, [this].concat(args))), _this), _this.handleClick = function (e) {
      _this.props.onClick(e);
      if (_this.props.isSelectable) {
        if (_this.props.isSelected) {
          _this.props.onDeselect();
        } else {
          _this.props.onSelect();
        }
      }
    }, _this.handleKeyPress = function (e) {
      if (_this.props.isSelectable) {
        if (e.key === 'Enter' || e.key === ' ') {
          _this.props.onSelect();
          e.preventDefault();
        }
      }

      _this.props.onKeyPress(e);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(TableRow, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          onClick = _props.onClick,
          onKeyPress = _props.onKeyPress,
          onSelect = _props.onSelect,
          onDeselect = _props.onDeselect,
          isHighlighted = _props.isHighlighted,
          isSelectable = _props.isSelectable,
          isSelected = _props.isSelected,
          _props$css = _props.css,
          css = _props$css === undefined ? {} : _props$css,
          props = _objectWithoutProperties(_props, ['children', 'onClick', 'onKeyPress', 'onSelect', 'onDeselect', 'isHighlighted', 'isSelectable', 'isSelected', 'css']);

      return React.createElement(
        Pane,
        _extends({
          display: 'flex'
        }, isSelectable ? {
          'aria-selected': isHighlighted,
          'aria-current': isSelected,
          css: _extends({}, selectableRowStyle, css),
          tabIndex: 0
        } : { css: css }, {
          onClick: this.handleClick,
          onKeyPress: this.handleKeyPress
        }, props),
        children
      );
    }
  }]);

  return TableRow;
}(PureComponent);

TableRow.propTypes = _extends({}, Pane.propTypes, {

  /**
   * Function that is called on click and enter/space keypress.
   */
  onSelect: PropTypes.func,

  /**
   * Function that is called on click and enter/space keypress.
   */
  onDeselect: PropTypes.func,

  /**
   * Makes the TableRow selectable.
   */
  isSelectable: PropTypes.bool,

  /**
   * Makes the TableRow selected.
   */
  isSelected: PropTypes.bool,

  /**
   * Manually set the TableRow to be highlighted.
   */
  isHighlighted: PropTypes.bool
});
TableRow.defaultProps = {
  onClick: function onClick() {},
  onSelect: function onSelect() {},
  onDeselect: function onDeselect() {},
  onKeyPress: function onKeyPress() {}
};
export default TableRow;
