// evergreen-ui/esm/table/src/TableHead.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/* eslint-disable react/jsx-handler-names */
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Pane } from '../../layers';
import ScrollbarSize from './ScrollbarSize';

var TableHead = function (_PureComponent) {
  _inherits(TableHead, _PureComponent);

  function TableHead() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, TableHead);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = TableHead.__proto__ || Object.getPrototypeOf(TableHead)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      scrollbarWidth: 0
    }, _this.handleScrollbarSize = function (width) {
      _this.setState({
        scrollbarWidth: width
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(TableHead, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          accountForScrollbar = _props.accountForScrollbar,
          props = _objectWithoutProperties(_props, ['children', 'accountForScrollbar']);

      var scrollbarWidth = this.state.scrollbarWidth;


      return React.createElement(
        Pane,
        _extends({ paddingRight: scrollbarWidth }, props),
        children,
        ' ',
        accountForScrollbar && React.createElement(ScrollbarSize, { handleScrollbarSize: this.handleScrollbarSize })
      );
    }
  }]);

  return TableHead;
}(PureComponent);

TableHead.propTypes = _extends({}, Pane.propTypes, {

  /**
   * This should always be true if you are using TableHead together with a TableBody.
   * Because TableBody has `overflowY: scroll` by default.
   */
  accountForScrollbar: PropTypes.bool
});
TableHead.defaultProps = {
  accountForScrollbar: true,
  boxSizing: 'border-box',
  display: 'flex',
  appearance: 'tint2',
  borderBottom: 'extraMuted'
};
export default TableHead;
