// evergreen-ui/commonjs/checkbox/src/Checkbox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _uiBox = require('ui-box');

var _uiBox2 = _interopRequireDefault(_uiBox);

var _typography = require('../../typography');

var _sharedStyles = require('../../shared-styles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var CheckIcon = function CheckIcon(_ref) {
  var _ref$fill = _ref.fill,
      fill = _ref$fill === undefined ? 'currentColor' : _ref$fill,
      props = _objectWithoutProperties(_ref, ['fill']);

  return _react2.default.createElement(
    'svg',
    _extends({ width: 10, height: 7, viewBox: '0 0 10 7' }, props),
    _react2.default.createElement('path', {
      fill: fill,
      fillRule: 'evenodd',
      d: 'M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z'
    })
  );
};

CheckIcon.propTypes = {
  fill: _propTypes2.default.string
};

var Checkbox = function (_PureComponent) {
  _inherits(Checkbox, _PureComponent);

  function Checkbox() {
    _classCallCheck(this, Checkbox);

    return _possibleConstructorReturn(this, (Checkbox.__proto__ || Object.getPrototypeOf(Checkbox)).apply(this, arguments));
  }

  _createClass(Checkbox, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          name = _props.name,
          label = _props.label,
          appearance = _props.appearance,
          disabled = _props.disabled,
          isInvalid = _props.isInvalid,
          checked = _props.checked,
          onChange = _props.onChange,
          value = _props.value,
          props = _objectWithoutProperties(_props, ['id', 'name', 'label', 'appearance', 'disabled', 'isInvalid', 'checked', 'onChange', 'value']);

      var appearanceStyle = _sharedStyles.CheckboxAppearances[appearance];

      return _react2.default.createElement(
        _uiBox2.default,
        _extends({
          is: 'label',
          cursor: disabled ? 'not-allowed' : 'pointer',
          display: 'flex',
          marginY: 16
        }, props),
        _react2.default.createElement(_uiBox2.default, _extends({
          is: 'input',
          id: id,
          type: 'checkbox',
          name: name,
          value: value,
          checked: checked,
          onChange: onChange,
          disabled: disabled
        }, isInvalid ? { 'aria-invalid': true } : {}, {
          css: appearanceStyle
        })),
        _react2.default.createElement(
          _uiBox2.default,
          {
            boxSizing: 'border-box',
            borderRadius: 3,
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            width: 16,
            height: 16
          },
          _react2.default.createElement(CheckIcon, null)
        ),
        label && _react2.default.createElement(
          _typography.Text,
          {
            marginLeft: 8,
            size: 300,
            color: disabled ? 'extraMuted' : 'default'
          },
          label
        )
      );
    }
  }]);

  return Checkbox;
}(_react.PureComponent);

Checkbox.propTypes = _extends({}, _uiBox2.default.propTypes, {
  id: _propTypes2.default.string,
  name: _propTypes2.default.string,
  label: _propTypes2.default.node,
  value: _propTypes2.default.string,
  checked: _propTypes2.default.bool,
  onChange: _propTypes2.default.func,
  disabled: _propTypes2.default.bool,
  isInvalid: _propTypes2.default.bool,
  appearance: _propTypes2.default.oneOf(Object.keys(_sharedStyles.CheckboxAppearances))
});
Checkbox.defaultProps = {
  appearance: 'default',
  onChange: function onChange() {}
};
exports.default = Checkbox;
