// evergreen-ui/esm/form-field/src/FormField.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box, { dimensions, spacing, position, layout } from 'ui-box';
import FormFieldLabel from './FormFieldLabel';
import FormFieldDescription from './FormFieldDescription';
import FormFieldValidationMessage from './FormFieldValidationMessage';

var FormField = function (_PureComponent) {
  _inherits(FormField, _PureComponent);

  function FormField() {
    _classCallCheck(this, FormField);

    return _possibleConstructorReturn(this, (FormField.__proto__ || Object.getPrototypeOf(FormField)).apply(this, arguments));
  }

  _createClass(FormField, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          label = _props.label,
          labelFor = _props.labelFor,
          children = _props.children,
          isRequired = _props.isRequired,
          labelProps = _props.labelProps,
          description = _props.description,
          validationMessage = _props.validationMessage,
          props = _objectWithoutProperties(_props, ['label', 'labelFor', 'children', 'isRequired', 'labelProps', 'description', 'validationMessage']);

      return React.createElement(
        Box,
        props,
        React.createElement(
          FormFieldLabel,
          _extends({
            htmlFor: labelFor,
            isAstrixShown: isRequired,
            marginBottom: 4
          }, labelProps),
          label
        ),
        children,
        validationMessage && React.createElement(
          FormFieldValidationMessage,
          null,
          validationMessage
        ),
        description && React.createElement(
          FormFieldDescription,
          { marginTop: 6 },
          description
        )
      );
    }
  }]);

  return FormField;
}(PureComponent);

FormField.propTypes = _extends({
  /**
   * The label used above the input element.
   */
  label: PropTypes.node.isRequired,

  /**
   * Passed on the label as a htmlFor prop.
   */
  labelFor: PropTypes.string,

  /**
   * Wether or not show a asterix after the label.
   */
  isRequired: PropTypes.bool,

  /**
   * A optional description of the field under the input element.
   */
  description: PropTypes.node,

  /**
   * If a validation message is passed it is shown under the input element
   * and above the description.
   */
  validationMessage: PropTypes.node

}, dimensions.propTypes, spacing.propTypes, position.propTypes, layout.propTypes);
FormField.defaultProps = {
  labelProps: {
    size: 400
  }
};
export default FormField;
