// evergreen-ui/esm/checkbox/src/Checkbox.js
var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import { Text } from '../../typography';
import { CheckboxAppearances } from '../../shared-styles';

var CheckIcon = function CheckIcon(_ref) {
  var _ref$fill = _ref.fill,
      fill = _ref$fill === undefined ? 'currentColor' : _ref$fill,
      props = _objectWithoutProperties(_ref, ['fill']);

  return React.createElement(
    'svg',
    _extends({ width: 10, height: 7, viewBox: '0 0 10 7' }, props),
    React.createElement('path', {
      fill: fill,
      fillRule: 'evenodd',
      d: 'M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z'
    })
  );
};

CheckIcon.propTypes = {
  fill: PropTypes.string
};

var Checkbox = function (_PureComponent) {
  _inherits(Checkbox, _PureComponent);

  function Checkbox() {
    _classCallCheck(this, Checkbox);

    return _possibleConstructorReturn(this, (Checkbox.__proto__ || Object.getPrototypeOf(Checkbox)).apply(this, arguments));
  }

  _createClass(Checkbox, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          id = _props.id,
          name = _props.name,
          label = _props.label,
          appearance = _props.appearance,
          disabled = _props.disabled,
          isInvalid = _props.isInvalid,
          checked = _props.checked,
          onChange = _props.onChange,
          value = _props.value,
          props = _objectWithoutProperties(_props, ['id', 'name', 'label', 'appearance', 'disabled', 'isInvalid', 'checked', 'onChange', 'value']);

      var appearanceStyle = CheckboxAppearances[appearance];

      return React.createElement(
        Box,
        _extends({
          is: 'label',
          cursor: disabled ? 'not-allowed' : 'pointer',
          display: 'flex',
          marginY: 16
        }, props),
        React.createElement(Box, _extends({
          is: 'input',
          id: id,
          type: 'checkbox',
          name: name,
          value: value,
          checked: checked,
          onChange: onChange,
          disabled: disabled
        }, isInvalid ? { 'aria-invalid': true } : {}, {
          css: appearanceStyle
        })),
        React.createElement(
          Box,
          {
            boxSizing: 'border-box',
            borderRadius: 3,
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            width: 16,
            height: 16
          },
          React.createElement(CheckIcon, null)
        ),
        label && React.createElement(
          Text,
          {
            marginLeft: 8,
            size: 300,
            color: disabled ? 'extraMuted' : 'default'
          },
          label
        )
      );
    }
  }]);

  return Checkbox;
}(PureComponent);

Checkbox.propTypes = _extends({}, Box.propTypes, {
  id: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.node,
  value: PropTypes.string,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  isInvalid: PropTypes.bool,
  appearance: PropTypes.oneOf(Object.keys(CheckboxAppearances))
});
Checkbox.defaultProps = {
  appearance: 'default',
  onChange: function onChange() {}
};
export default Checkbox;
