// evergreen-ui/esm/typography/src/Paragraph.js
var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

import React, { PureComponent } from 'react';
import mapValues from 'lodash.mapvalues';
import TextStyles from './styles/TextStyles';
import Text from './Text';

var textStyleTransformation = function textStyleTransformation(_ref) {
  var lineHeight = _ref.lineHeight,
      textStyle = _objectWithoutProperties(_ref, ['lineHeight']);

  return _extends({}, textStyle, {
    // Multiply line height by 1.1
    lineHeight: Math.round(parseFloat(lineHeight, 10) * 1.08) + 'px'
  });
};

var textStyles = mapValues(TextStyles, textStyleTransformation);

var Paragraph = function (_PureComponent) {
  _inherits(Paragraph, _PureComponent);

  function Paragraph() {
    _classCallCheck(this, Paragraph);

    return _possibleConstructorReturn(this, (Paragraph.__proto__ || Object.getPrototypeOf(Paragraph)).apply(this, arguments));
  }

  _createClass(Paragraph, [{
    key: 'render',
    value: function render() {
      return React.createElement(Text, _extends({
        is: 'p',
        marginTop: 0,
        marginBottom: 0,
        textStyles: textStyles
      }, this.props));
    }
  }]);

  return Paragraph;
}(PureComponent);

Paragraph.propTypes = _extends({}, Text.propTypes);
export default Paragraph;
