// evergreen-ui/esm/typography/src/Text.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import FontFamilies from './styles/FontFamilies';
import TextStyles from './styles/TextStyles';
import TextUppercaseStyles from './styles/TextUppercaseStyles';
import TextColors from './styles/TextColors';

var isDev = process.env.NODE_ENV === 'development' || !process.env.NODE_ENV;

var Text = function (_PureComponent) {
  _inherits(Text, _PureComponent);

  function Text() {
    _classCallCheck(this, Text);

    return _possibleConstructorReturn(this, (Text.__proto__ || Object.getPrototypeOf(Text)).apply(this, arguments));
  }

  _createClass(Text, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          size = _props.size,
          color = _props.color,
          textStyles = _props.textStyles,
          textUppercaseStyles = _props.textUppercaseStyles,
          fontFamily = _props.fontFamily,
          isUppercase = _props.isUppercase,
          props = _objectWithoutProperties(_props, ['size', 'color', 'textStyles', 'textUppercaseStyles', 'fontFamily', 'isUppercase']);

      var textStyle = textStyles[size];

      if (isUppercase) {
        // Only 100 and 200 support uppercase styles atm
        // Fallback on non uppercase atm
        if (Object.prototype.hasOwnProperty.call(textUppercaseStyles, size)) {
          textStyle = textUppercaseStyles[size];
        } else if (isDev) {
          console.error('Uppercase ' + size + ' not supported. <Text isUppercase> only supports the following sizes: ' + JSON.stringify(Object.keys(textUppercaseStyles).map(Number)));
        }
      }

      return React.createElement(Box, _extends({
        is: 'span'
      }, color ? { color: TextColors[color] || color } : {}, {
        fontFamily: FontFamilies[fontFamily] || fontFamily
      }, textStyle, props));
    }
  }]);

  return Text;
}(PureComponent);

Text.propTypes = _extends({}, Box.propTypes, {

  /**
   * Size of the text style.
   * Can be: 100, 200, 300, 400, 500, 600, 700, 800, 900.
   */
  size: PropTypes.oneOf(Object.keys(TextStyles).map(Number)),

  /**
   * Font family.
   * Can be: ui, display or mono
   */
  fontFamily: PropTypes.oneOf(Object.keys(FontFamilies)),

  /**
   * Sets the text to uppercase.
   * Only sizes 100 and 200 support uppercase styles at the moment.
   */
  isUppercase: PropTypes.bool,

  /**
   * The text styles available.
   * This is overridden by other text components that implement Text as the base.
   * You should avoid setting this manually.
   */
  textStyles: PropTypes.object,

  /**
   * The uppercase text styles.
   * You should avoid setting this manually.
   */
  textUppercaseStyles: PropTypes.object
});
Text.defaultProps = {
  size: 500,
  color: 'default',
  fontFamily: 'ui',
  textStyles: TextStyles,
  textUppercaseStyles: TextUppercaseStyles
};
export default Text;
