// evergreen-ui/esm/toaster/src/Toaster.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

import React from 'react';
import ReactDOM from 'react-dom';
import ToastManager from './ToastManager';

var ID = 'evergreen-toaster';
var isBrowser = typeof window !== 'undefined' && typeof window.document !== 'undefined';

/**
 * The Toaster manages the interactionsb between
 * the ToasterManger and the toast API.
 */

var Toaster = function Toaster() {
  var _this = this;

  _classCallCheck(this, Toaster);

  this._bindNotify = function (handler) {
    _this.notifyHandler = handler;
  };

  this._bindGetToasts = function (handler) {
    _this.getToastsHandler = handler;
  };

  this._bindCloseAll = function (handler) {
    _this.closeAllHandler = handler;
  };

  this.getToasts = function () {
    return _this.getToastsHandler();
  };

  this.closeAll = function () {
    return _this.closeAllHandler();
  };

  this.notify = function (title) {
    var settings = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    return _this.notifyHandler(title, _extends({}, settings, { type: 'default' }));
  };

  this.success = function (title) {
    var settings = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    return _this.notifyHandler(title, _extends({}, settings, { type: 'success' }));
  };

  this.warning = function (title) {
    var settings = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    return _this.notifyHandler(title, _extends({}, settings, { type: 'warning' }));
  };

  this.danger = function (title) {
    var settings = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    return _this.notifyHandler(title, _extends({}, settings, { type: 'danger' }));
  };

  this.info = function (title) {
    var settings = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    return _this.notifyHandler(title, _extends({}, settings, { type: 'info' }));
  };

  if (!isBrowser) return;
  var container = void 0;

  var element = document.getElementById(ID);
  if (element) {
    container = element;
  } else {
    // Create container if not exists yet.
    container = document.createElement('div');
    container.id = ID;
    document.body.appendChild(container);
  }

  ReactDOM.render(React.createElement(ToastManager, {
    bindNotify: this._bindNotify,
    bindGetToasts: this._bindGetToasts,
    bindCloseAll: this._bindCloseAll
  }), container);
};

export default Toaster;
