// evergreen-ui/esm/file-picker/src/FilePicker.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import { Button } from '../../buttons';
import { TextInput } from '../../text-input';

export var CLASS_PREFIX = 'evergreen-file-picker';

var FilePicker = function (_PureComponent) {
  _inherits(FilePicker, _PureComponent);

  function FilePicker() {
    _classCallCheck(this, FilePicker);

    var _this = _possibleConstructorReturn(this, (FilePicker.__proto__ || Object.getPrototypeOf(FilePicker)).call(this));

    _this.fileInputRef = function (node) {
      _this.fileInput = node;
    };

    _this.handleFileChange = function (e) {
      var onChange = _this.props.onChange;
      // Firefox returns the same array instance each time for some reason

      var files = [].concat(_toConsumableArray(e.target.files));

      _this.setState({ files: files });

      if (onChange) {
        onChange(files);
      }
    };

    _this.handleButtonClick = function () {
      _this.fileInput.click();
    };

    _this.state = {
      files: []
    };
    return _this;
  }

  _createClass(FilePicker, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          name = _props.name,
          accept = _props.accept,
          required = _props.required,
          multiple = _props.multiple,
          disabled = _props.disabled,
          capture = _props.capture,
          height = _props.height,
          onChange = _props.onChange,
          props = _objectWithoutProperties(_props, ['name', 'accept', 'required', 'multiple', 'disabled', 'capture', 'height', 'onChange']);

      var files = this.state.files;


      var inputValue = void 0;
      if (files.length === 0) {
        inputValue = '';
      } else if (files.length === 1) {
        inputValue = files[0].name;
      } else {
        inputValue = files.length + ' files';
      }

      var buttonText = void 0;
      if (files.length === 0) {
        buttonText = 'Select file';
      } else if (files.length === 1) {
        buttonText = 'Replace file';
      } else {
        buttonText = 'Replace files';
      }

      return React.createElement(
        Box,
        _extends({ display: 'flex', className: CLASS_PREFIX + '-root' }, props),
        React.createElement(Box, {
          innerRef: this.fileInputRef,
          className: CLASS_PREFIX + '-file-input',
          is: 'input',
          type: 'file',
          name: name,
          accept: accept,
          required: required,
          multiple: multiple,
          disabled: disabled,
          capture: capture,
          onChange: this.handleFileChange,
          display: 'none'
        }),
        React.createElement(TextInput, {
          className: CLASS_PREFIX + '-text-input',
          readOnly: true,
          value: inputValue,
          placeholder: 'Select a file to upload\u2026'
          // There's a weird specifity issue when there's two differently sized inputs on the page
          , borderTopRightRadius: '0 !important',
          borderBottomRightRadius: '0 !important',
          height: height,
          flex: 1,
          textOverflow: 'ellipsis'
        }),
        React.createElement(
          Button,
          {
            className: CLASS_PREFIX + '-button',
            onClick: this.handleButtonClick,
            disabled: disabled,
            borderTopLeftRadius: 0,
            borderBottomLeftRadius: 0,
            height: height,
            flexShrink: 0
          },
          buttonText
        )
      );
    }
  }]);

  return FilePicker;
}(PureComponent);

FilePicker.propTypes = {
  name: PropTypes.string,
  accept: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  required: PropTypes.bool,
  multiple: PropTypes.bool,
  disabled: PropTypes.bool,
  capture: PropTypes.bool,
  height: PropTypes.number,
  onChange: PropTypes.func
};
export default FilePicker;
