// evergreen-ui/commonjs/ssr/src/extractStyles.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = extractStyles;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _uiBox = require('ui-box');

var _server = require('glamor/server');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function extractStyles() {
  var _boxExtractStyles = (0, _uiBox.extractStyles)(),
      styles = _boxExtractStyles.styles,
      cache = _boxExtractStyles.cache;

  var _renderStatic = (0, _server.renderStatic)(function () {
    return 'let glamor believe there is some html here';
  }),
      ids = _renderStatic.ids,
      css = _renderStatic.css;

  var evergreenCache = {
    uiBoxCache: cache,
    glamorIds: ids
  };

  var scriptProps = {
    type: 'application/json',
    id: 'evergreen-hydrate',
    dangerouslySetInnerHTML: { __html: JSON.stringify(evergreenCache) }
  };

  return {
    css: styles + '\n' + css,
    cache: evergreenCache,
    hydrationScript: _react2.default.createElement('script', scriptProps)
  };
}
