// evergreen-ui/esm/alert/src/InlineAlert.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { spacing, dimensions, position, layout } from 'ui-box';
import { colors } from '../../colors';
import { Pane } from '../../layers';
import { Text } from '../../typography';
import { CheckCircleIcon, DangerIcon, QuestionIcon, WarningIcon } from '../../icons';

var getColorForType = function getColorForType(type) {
  switch (type) {
    case 'success':
      return colors.green['500'];
    case 'question':
    default:
      return colors.blue['500'];
    case 'danger':
      return colors.red['500'];
    case 'warning':
      return colors.yellow['500'];
  }
};

var getIconForType = function getIconForType(type) {
  var iconProps = {
    size: 22,
    iconSize: 14,
    color: getColorForType(type)
  };

  switch (type) {
    case 'success':
      return React.createElement(CheckCircleIcon, iconProps);
    case 'question':
    default:
      return React.createElement(QuestionIcon, iconProps);
    case 'danger':
      return React.createElement(DangerIcon, iconProps);
    case 'warning':
      return React.createElement(WarningIcon, iconProps);
  }
};

var InlineAlert = function (_PureComponent) {
  _inherits(InlineAlert, _PureComponent);

  function InlineAlert() {
    _classCallCheck(this, InlineAlert);

    return _possibleConstructorReturn(this, (InlineAlert.__proto__ || Object.getPrototypeOf(InlineAlert)).apply(this, arguments));
  }

  _createClass(InlineAlert, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          type = _props.type,
          hasIcon = _props.hasIcon,
          size = _props.size,
          props = _objectWithoutProperties(_props, ['children', 'type', 'hasIcon', 'size']);

      return React.createElement(
        Pane,
        _extends({ alignItems: 'center', display: 'flex' }, props),
        hasIcon && type !== 'default' && React.createElement(
          Pane,
          { display: 'inline', marginRight: 8 },
          getIconForType(type)
        ),
        React.createElement(
          Text,
          { size: size, fontWeight: 500 },
          children
        )
      );
    }
  }]);

  return InlineAlert;
}(PureComponent);

InlineAlert.propTypes = _extends({}, spacing.propTypes, position.propTypes, layout.propTypes, dimensions.propTypes, {

  /**
   * The content of the alert.
   */
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),

  /**
   * The type of the alert. This should always be set explicitly.
   */
  type: PropTypes.oneOf(['default', 'success', 'warning', 'danger', 'question']),

  /**
   * When true, show a icon on the left matching the type.
   * There is no point not showing this.
   */
  hasIcon: PropTypes.bool,

  /**
   * The size of the Text.
   */
  size: PropTypes.number
});
InlineAlert.defaultProps = {
  type: 'default',
  hasIcon: true,
  size: 400
};
export default InlineAlert;
